# WHAT ARE THE MOST USED APPLICATIONS ?

In their **daily activities**, individuals used to use some mobile, web as well as desktop applications. The fact is that they tend to use some more than others. 

Here, I found on Kaggle, a dataset that contains some data about people and their usage rate of some common applications. 

### Let's do a quick Exploratory Data Analysis to see what conclusions can be drawn: 
  
1. Which applications are completely free ?
2. How much does the most expensive application cost ?
3. Which application received the most important number of reviews ?
4. What are the best rated applications ?

## Importing the necessary libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

## Display the content of the dataset

In [2]:
common_apps_data = pd.read_csv("./common_apps.csv")
common_apps_data

App_Order                                           App_Name  App_Star  \
0            0                                             iTunes       3.0   
1            1                                            Netflix       4.0   
2            2                                      Spotify Music       4.0   
3            3                             Microsoft Sticky Notes       4.5   
4            4                                           Facebook       3.5   
..         ...                                                ...       ...   
992        992                              HP System Information       3.5   
993        993  Any Sound Recorder: audio recorder &amp; voice...       4.0   
994        994                                       Kiddo Smart!       4.5   
995        995                                           TowMagic       NaN   
996        996                              ISO Image Creator PRO       3.5   

    App_Price  App_Review  
0        Free     30874.0  
1        Free    671575.0  
2        Free     31093.0  
3        Free     90236.0  
4        Free    481943.0  
..        ...         ...  
992      Free        21.0  
993      Free       173.0  
994      Free       524.0  
995      Free         NaN  
996      Free       138.0  

[997 rows x 5 columns]

According to the display dataset, it can obviously be seen that all the data are not shown. 

## Display the whole dataset
There is a lot of records in the dataset so it will be kind of very long displaying as it contains 997 rows.

In [3]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', 5)
common_apps_data

App_Order                                           App_Name  App_Star  \
0            0                                             iTunes       3.0   
1            1                                            Netflix       4.0   
2            2                                      Spotify Music       4.0   
3            3                             Microsoft Sticky Notes       4.5   
4            4                                           Facebook       3.5   
5            5                                          Instagram       3.0   
6            6                                               Hulu       3.0   
7            7                                          Messenger       3.0   
8            8                                         Your Phone       4.0   
9            9                                            OneNote       4.5   
10          10                                   WhatsApp Desktop       3.5   
11          11                                    Norton Safe Web       3.5   
12          12                                              Slack       4.0   
13          13                               NVIDIA Control Panel       2.0   
14          14                                        Xbox (Beta)       2.5   
15          15                                             iCloud       2.5   
16          16                                  Snip &amp; Sketch       3.0   
17          17                                            Pandora       4.0   
18          18                       Grammarly for Microsoft Edge       3.5   
19          19      Microsoft To Do: Lists, Tasks &amp; Reminders       4.5   
20          20                                           Paint 3D       4.5   
21          21                            Any Player - Video, DVD       4.5   
22          22  Adobe Photoshop Express: Image Editor, Adjustm...       4.5   
23          23  Trio Office: Word, Slide, Spreadsheet &amp; PD...       4.5   
24          24  Cool File Viewer: Rar, Word, PDF, PPT, Video &...       4.5   
25          25                                              Honey       4.0   
26          26                                                VLC       3.5   
27          27                     PDF Reader - View, Edit, Share       4.5   
28          28                                          Pinterest       5.0   
29          29                                 EasyMail for Gmail       4.5   
30          30                                             Office       3.5   
31          31                                       Amazon Music       3.5   
32          32                                  DTS Sound Unbound       4.5   
33          33                              VitalSource Bookshelf       3.5   
34          34               BreeZip: Rar, Zip &amp; 7z Extractor       4.5   
35          35                                   Power BI Desktop       4.0   
36          36                            Hotspot Shield Free VPN       4.0   
37          37                          FizzTube - YouTube Player       4.5   
38          38                                   Telegram Desktop       4.5   
39          39                                              Alexa       2.5   
40          40                                             Fitbit       3.5   
41          41               Drawboard PDF - Read, edit, annotate       4.0   
42          42                                       Dolby Access       4.5   
43          43                                Autodesk SketchBook       4.5   
44          44                           Microsoft Remote Desktop       3.5   
45          45                                 WeChat For Windows       3.0   
46          46                                 Windows DVD Player       3.0   
47          47                            Animotica - Movie Maker       4.5   
48          48                                       Feedback Hub       3.0   
49          49                                    Norto

## How many records exist in the dataset (number of lines and columns) ?

In [4]:
common_apps_data.shape

(997, 5)

The data contains 997 rows and 5 columns. Quite a large dataset but let's see if all the records (lines and columns) are present.

## What are the different columns of the dataset ?

In [5]:
common_apps_data.columns.tolist()

['App_Order', 'App_Name', 'App_Star', 'App_Price', 'App_Review']

So these 5 elements represent the different columns of the dataset. In the order, lets'name : 
- **App_Order** for the order of the applications in the dataset
- **App_Name** for the names of the applications
- **App_Star** for the application rating stars
- **App_Price** for the applications prices and 
- **App_Review** for the different reviews made on the applications.

# Now, let's find an answer to the different questions

## 1. Which applications are completely free ?

In [6]:
pd.set_option('display.max_rows', None)
free_apps = common_apps_data[common_apps_data["App_Price"]=="Free"]
free_apps

App_Order                                           App_Name  App_Star  \
0            0                                             iTunes       3.0   
1            1                                            Netflix       4.0   
2            2                                      Spotify Music       4.0   
3            3                             Microsoft Sticky Notes       4.5   
4            4                                           Facebook       3.5   
5            5                                          Instagram       3.0   
6            6                                               Hulu       3.0   
7            7                                          Messenger       3.0   
8            8                                         Your Phone       4.0   
9            9                                            OneNote       4.5   
10          10                                   WhatsApp Desktop       3.5   
11          11                                    Norton Safe Web       3.5   
12          12                                              Slack       4.0   
13          13                               NVIDIA Control Panel       2.0   
14          14                                        Xbox (Beta)       2.5   
15          15                                             iCloud       2.5   
16          16                                  Snip &amp; Sketch       3.0   
17          17                                            Pandora       4.0   
18          18                       Grammarly for Microsoft Edge       3.5   
19          19      Microsoft To Do: Lists, Tasks &amp; Reminders       4.5   
20          20                                           Paint 3D       4.5   
21          21                            Any Player - Video, DVD       4.5   
22          22  Adobe Photoshop Express: Image Editor, Adjustm...       4.5   
23          23  Trio Office: Word, Slide, Spreadsheet &amp; PD...       4.5   
24          24  Cool File Viewer: Rar, Word, PDF, PPT, Video &...       4.5   
25          25                                              Honey       4.0   
26          26                                                VLC       3.5   
27          27                     PDF Reader - View, Edit, Share       4.5   
28          28                                          Pinterest       5.0   
29          29                                 EasyMail for Gmail       4.5   
30          30                                             Office       3.5   
31          31                                       Amazon Music       3.5   
32          32                                  DTS Sound Unbound       4.5   
33          33                              VitalSource Bookshelf       3.5   
34          34               BreeZip: Rar, Zip &amp; 7z Extractor       4.5   
35          35                                   Power BI Desktop       4.0   
36          36                            Hotspot Shield Free VPN       4.0   
37          37                          FizzTube - YouTube Player       4.5   
38          38                                   Telegram Desktop       4.5   
39          39                                              Alexa       2.5   
40          40                                             Fitbit       3.5   
41          41               Drawboard PDF - Read, edit, annotate       4.0   
42          42                                       Dolby Access       4.5   
43          43                                Autodesk SketchBook       4.5   
44          44                           Microsoft Remote Desktop       3.5   
45          45                                 WeChat For Windows       3.0   
46          46                                 Windows DVD Player       3.0   
47          47                            Animotica - Movie Maker       4.5   
48          48                                       Feedback Hub       3.0   
49          49                                    Norto

## Let's count the number of the free applications recorded in the dataset

In [7]:
pd.set_option('display.max_rows', None)
free_apps = common_apps_data[common_apps_data["App_Price"]=="Free"]
free_apps.count(axis=0)

App_Order     907
App_Name      907
App_Star      885
App_Price     907
App_Review    885
dtype: int64

According to the **App_Price** count, it can be inferred that 907 out of 997, approximately 91% of the applications are used for free. 

### What about the applications that are not free ?

In [8]:
pd.set_option('display.max_rows', None)
not_free_apps = common_apps_data[common_apps_data["App_Price"]!="Free"]
not_free_apps

App_Order                                           App_Name  App_Star  \
630        630                           Duplicates Cleaner Prime       5.0   
631        631                            Video Maker - VideoShow       4.0   
632        632                                          iThoughts       4.0   
633        633                                       Sketch Board       4.0   
634        634                                         NetTV Plus       3.0   
635        635                   Business Card Maker And Designer       5.0   
636        636                             Learn Typing in 1 Hour       5.0   
637        637                                 Food Cost Profiler       5.0   
638        638                         Pages Manager for Facebook       3.5   
639        639                                           GoldWave       3.5   
640        640                WiFi Explore - SpeedTest, WiFi Scan       4.0   
641        641                                      Aviation Exam       4.5   
642        642            Intel(R) Management and Security Status       3.5   
643        643                                     Nextgen Reader       4.5   
644        644                                         Oracle MWM       5.0   
645        645                                        Sumatra PDF       4.5   
646        646  JUMPING VPN – Unlimited Free VPN &amp; Fast Se...       4.5   
647        647                                Corel PaintShop Pro       3.0   
648        648                               Screen Recorder Plus       4.0   
649        649                                    Color Explosion       4.0   
650        650                                        Text Reader       4.0   
651        651                                Document Translator       4.0   
652        652                   MixPad Multitrack Recording Free       4.0   
653        653                                 Spending Tracker ™       4.5   
654        654  Monument Browser - Ad Blocker, Privacy Focused...       4.0   
655        655                                     Media Importer       3.0   
656        656                                       Baby Animals       4.0   
657        657                    KONICA MINOLTA Print Experience       2.0   
658        658                                     Flashcard Hero       4.0   
659        659                                Sentence Diagrammer       2.0   
660        660                                           Power BI       4.0   
661        661                                  Live TV Ultimated       4.5   
662        662                                   Samsung Settings       2.0   
663        663                                     WinDbg Preview       4.0   
664        664                   PDF Manager - Merge, Split, Trim       4.0   
665        665                                 Json Formatter Pro       4.0   
666        666                               Collector for ArcGIS       4.0   
667        667                                      Discord Tiles       3.5   
668        668                                            TestNav       2.0   
669        669                                 Express Scribe Pro       5.0   
670        670               Video Editor &amp; Music Movie Maker       4.0   
671        671                            Try a Little Tenderness       4.0   
672        672                                Easy File Organizer       4.5   
673        673                               inPixio Photo Editor       3.5   
674        674                                    Eclipse Manager       4.0   
675        675  Suite for MS Office - Templates for Microsoft ...       2.0   
676        676                                              Fotor       4.0   
677        677  8 Zip Pro - advanced archiver for Zip, Rar, 7Z...       4.5   
678        678                                        Token2Shell       4.0   
679        679                        Dashlane - Passwo

### Counting the number of applications that are not free (Just to be sure it remains the 90)

In [9]:
not_free_apps.count(axis=0)

App_Order     90
App_Name      90
App_Star      87
App_Price     90
App_Review    87
dtype: int64

Under the **App_Price**, it shows 90, which means that 90 applications are not free.

## 2. How much does the most expensive application cost ?

In [16]:
most_exp_app_price = not_free_apps["App_Price"].max()
most_exp_app_price

'$5.29 '

From one of the last analysis, all the applications that are **not free** cost $5.29.

So, when querying the data to find out the price of the most expensive application, it displays $5.29. 

**Conclusion** : More than one applications (90) have the same price ($5.29) which means that there is no single most expensive application. 

## 3. Which application received the most important number of reviews ?

In [56]:
no_of_reviews = common_apps_data["App_Review"].max()
no_of_reviews

671575.0

In [57]:
most_rev_apps = common_apps_data[["App_Name","App_Review"]].query("App_Review == 671575.0")
most_rev_apps

App_Name  App_Review
1  Netflix    671575.0

The maximum number of reviews made by users is observed on **Netflix**. 

## 4. What are the best rated applications ?

In [60]:
pd.set_option('display.max_rows', None)
rated_apps = common_apps_data[["App_Name","App_Star"]].query("App_Star == 5.0")
rated_apps

App_Name  App_Star
28                                           Pinterest       5.0
73                                               PDF X       5.0
87                                  Beach Time PREMIUM       5.0
155                                         Beach Glow       5.0
156                                    Snap Core First       5.0
245                                         Beach Glow       5.0
246                                    Snap Core First       5.0
286                                OMEN Command Center       5.0
289                                      Dogs and Cats       5.0
300                  Cricut Design Space For Beginners       5.0
302                                 DX1 for Windows 10       5.0
347                              Ultimate Photo Editor       5.0
368  Logo Maker - Logo Creator, Generator &amp; Des...       5.0
383                       Password Manager SafeInCloud       5.0
412                                River Delta PREMIUM       5.0
431      WinSCP - SFTP, FTP, WebDAV, SCP and S3 client       5.0
446  Poster Maker: Thumbnail, Cover, Flyer &amp; Ad...       5.0
458                                  What Next PREMIUM       5.0
465                                     Chilly Morning       5.0
484                   CyberLink PhotoDirector 11 Ultra       5.0
489                          Trains on Bridges PREMIUM       5.0
491                               Fast Weigh Ticketing       5.0
493                              Winding Roads PREMIUM       5.0
498  Music Editor : Trim, Extract, Convert and Mix ...       5.0
500                                           Snipaste       5.0
506                                Fractal Art PREMIUM       5.0
545                               Anime Unlimited 2019       5.0
603                               Sun and Sand PREMIUM       5.0
606    Poster Maker, Flyer Designer, Ads Page Designer       5.0
623                                    Day on the Farm       5.0
626                                   Free Movies # HD       5.0
630                           Duplicates Cleaner Prime       5.0
635                   Business Card Maker And Designer       5.0
636                             Learn Typing in 1 Hour       5.0
637                                 Food Cost Profiler       5.0
644                                         Oracle MWM       5.0
669                                 Express Scribe Pro       5.0
699     Splashtop Business – Remote Support and Access       5.0
708                             Cartoons Unlimited Pro       5.0
709                                           PhotoPad       5.0
728                                   Outrush Ordinary       5.0
732               Photo Cut Paste : Background Changer       5.0
735                                 AdBlocker Ultimate       5.0
744                     The Grand Canyon National Park       5.0
760                             inPixio Photo Clip Pro       5.0
773                                    CrystalDiskMark       5.0
778                                  iBackup Extractor       5.0
788                         Crescendo Master's Edition       5.0
795                                    TouchMD Consult       5.0
867                                           CalenGoo       5.0
895                                        Express Zip       5.0
918                                           CDK Lane       5.0
931                     Ultimate Photo Blender / Mixer       5.0
937                                         Frosty Art       5.0
945                                           pcdMagic       5.0

In [64]:
rated_apps["App_Name"].count()

55

**55** applications out of 997 have been appreciated by the users, probably due to their unique user experience and the communication strategy.